src/cloud.c

```diff
@@ -426,88 +418,102 @@ int cloud_publish_data(const char *id, uint8_t sensor_id, uint8_t value_type,
 		       const knot_value_type *value,
 		       uint8_t kval_len)
 {
-	amqp_bytes_t queue_cloud;
 	json_object *jobj_data;
 	const char *json_str;
 	int result;
 
-	queue_cloud = mq_declare_new_queue(MQ_QUEUE_FOG_IN);
-	if (!queue_cloud.bytes) {
-		l_error("Error on declare a new queue");
-		return -1;
-	}
-
 	jobj_data = parser_data_create_object(id, sensor_id, value_type, value,
 					      kval_len);
-	if (!jobj_data) {
-		amqp_bytes_free(queue_cloud);
+	if (!jobj_data)
 		return KNOT_ERR_CLOUD_FAILURE;
-	}
+
 	json_str = json_object_to_json_string(jobj_data);
 
 	headers[0].value.value.bytes = amqp_cstring_bytes(user_auth_token);
 
-	result = mq_publish_persistent_message(queue_cloud,
-					       MQ_EXCHANGE_FOG_IN,
-					       MQ_CMD_DATA_PUBLISH,
-					       headers, 1,
-					       MQ_MSG_EXPIRATION_TIME_MS,
-					       json_str);
+	result = mq_publish_fanout_message(MQ_EXCHANGE_DATA_SENT,
+					   headers, 1,
```

Comment:
Add a comment informing whats is 1 or why it should be 1.

Reply:
I put a disclaimer informing about the AMQP settings of the sent message